weblate/static/editor/tools/search.js

```diff
@@ -0,0 +1,153 @@
+                $searchPreview.text(gettext("No results found"));
+              } else {
+                showResults(results, $searchElment);
+              }
+            },
+          });
+        }
+      }, 300); // If the user stops typing for 300ms, the search results will be fetched
+    });
+
+    // Show the preview on focus
+    $searchElment.on("focus", () => {
+      if ($searchElment.val() !== "" && $searchPreview.html() !== "") {
+        $searchPreview.show();
+        $("#results-num").show();
+      }
+    });
+
+    // Close the preview on focusout, form submit, form reset, and form clear
+    $searchElment.on("focusout", () => {
+      // Hide after a delay to allow click on a result
+      setTimeout(() => {
+        $searchPreview.hide();
+        $("#results-num").hide();
+      }, 700);
+    });
+    $searchForm.on("submit", () => {
+      $searchPreview.html("");
+      $searchPreview.hide();
+      $("#results-num").remove();
+    });
+    $searchForm.on("reset", () => {
+      $searchPreview.html("");
+      $searchPreview.hide();
+      $("#results-num").remove();
+    });
+    $searchForm.on("clear", () => {
+      $searchPreview.html("");
+      $("#results-num").remove();
+      $searchPreview.hide();
+    });
+
+    /**
+     * Handles the search results and displays them in the preview element.
+     * @param {any} results fetched search results
+     * @param {jQuery} $searchElment The search input element
+     * @returns void
+     */
+    function showResults(results, $searchElment) {
+      // Show the number of results
+      if (results.length > 0) {
+        $searchElment.after(
+          `<h4 id="results-num">${gettext("Found: ")} ${results.length}</h4>`,
+        );
+      } else {
+        $("#results-num").remove();
+      }
+
+      for (const result of results) {
+        const key = result.context;
+        const source = result.source;
+
+        // Make the URL relative
+        const url = result.web_url.replace(/^[a-zA-Z]+:\/\/[^/]+\//, "/");
+        const resultHtml = `
+          <a href="${url}" target="_blank" id="search-result" rel="noopener noreferrer">
+            <small>${key}</small>
+            <div>
+              ${source.toString()}
+            </div>
+          </a>
+        `;
+
+        $searchPreview.append(resultHtml);
+      }
+    }
+  }
+
+  /**
+   * Builds a search query string from the user input filters.
+   * The search query is built by the user input filters.
+   * The path lookup is also added to the search query.
+   * Built in the following format: `path:proj/comp filters`.
+   *
+   * @param {jQuery} $searchElment - The user input.
+   * @returns {string} - The built search query string.
+   *
+   * */
+  function buildSearchQuery($searchElment) {
+    let builtSearchQuery = "";
+
+    // Add path lookup to the search query
+    const pathSplited = window.location.pathname.split("/");
+    const projectPath = `${pathSplited[2]}/${pathSplited[3]}`;
```

Comment:
This is wrong - it strips language and it will break on categorized components. Instead of guessing it from location (what is easy to break in the future), let's push it from backend. I've done that in https://github.com/WeblateOrg/weblate/pull/12602, you can then get the path from the form.